### ISSUE TITLE
AMQPParser throw an error: Oversize frame 131318

### ISSUE THREAD

**Issue by Vaidas737** [dropbox/amqp-coffee#70]
I got the following error after having a lot of rejects.
It may be that putting a message in a dead-letter queue adds things to the message header, and doing this repeatedly makes the message too large.

I need to be able to handle this kind of exception, but connection 'error' event is not firing on this kind of error.

2016-11-22T18:10:34+0000[ID=1] events.js:72
2016-11-22T18:10:34+0000[ID=1]         throw er; // Unhandled 'error' event
2016-11-22T18:10:34+0000[ID=1]               ^
2016-11-22T18:10:34+0000[ID=1] Error: Oversize frame 131318
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.error (/usr/local/logdog/card-guard-processor/node_
modules/amqp-coffee/bin/src/lib/AMQPParser.js:150:21)
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.header (/usr/local/logdog/card-
guard-processor/node_modules/amqp-coffee/bin/src/lib/AMQPParser.js:46:23)
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.frameEnd (/usr/local/logdog/card-guard-processor/node_modules/amqp-coffee/bin/src/lib/AMQPParser.js:95:19)
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.frame (/usr/local/logdog/card-guard-processor/node_modules/amqp-coffee/bin/src/lib/AMQPParser.js:64:21)
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.header (/usr/local/logdog/card-guard-processor/node_modules/amqp-coffee/bin/src/lib/AMQPParser.js:51:21)
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.frameEnd (/usr/local/logdog/card-guard-processor/node_modules/amqp-coffee/bin/src/lib/AMQPParser.js:95:19)
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.frame (/usr/local/logdog/card-guard-processor/node_modules/amqp-coffee/bin/src/lib/AMQPParser.js:64:21)
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.header (/usr/local/logdog/card-guard-processor/node_modules/amqp-coffee/bin/src/lib/AMQPParser.js:51:21)
2016-11-22T18:10:34+0000[ID=1]     at AMQPParser.execute (/usr/local/logdog/card-guard-processor/node_modules/amqp-coffee/bin/src/lib/AMQPParser.js:31:33)
2016-11-22T18:10:34+0000[ID=1]     at Socket.<anonymous> (/usr/local/logdog/card-guard-processor/node_modules/amqp-coffee/bin/src/lib/Connection.js:497:29)

**Comment by witq** [comment]
Maybe the event is emitted not by the connection but by the consumer? I may be completely wrong, but you can try that.

**Comment by barshow** [comment]
The frame parser was throwing the error so you could watch connection.parser.on 'error' however that is somewhat broken, so I switched it to bubble the error up to the connection

https://github.com/dropbox/amqp-coffee/pull/74 

This will be fixed in 0.1.30 which will be published within the next day or two.  Thanks for the report!

**Comment by Vaidas737** [comment]
Thanks!

**Comment by Vaidas737** [comment]
Just making sure - what will happen to the connection to rabbitmq after parser error? The connection will still alive and it's fine to keep working? Or some bad things may happen and reconnection is required?

**Comment by barshow** [comment]
@Vaidas737 I changed my thinking on this a little.

I think the connection needs to be destroyed, and ignoring a oversize frame shouldn't be an option.  Here is my reasoning.  A oversized from should never exist if it does then the AMQP broker has a bug, or the client has a bug and has interpreted a packet or partial packet incorrectly. To fix that the connection would need to be restarted.  If we ignore a oversize frame we have no way of knowing if the data is truly that size and what to do if its not.  I think throwing a error and killing the connection is the proper thing to do.  So I updated the pr to reflect this, let me know if you disagree
